test/tst_filepath.py

```diff
@@ -0,0 +1,15 @@
+import os
+import unittest
+import netCDF4
+
+class test_filepath(unittest.TestCase):
+
+    def setUp(self):
+        self.netcdf_file = os.path.join(os.getcwd(), "netcdf_dummy_file.nc")
+        self.nc = netCDF4.Dataset(self.netcdf_file)
+
+    def test_filepath(self):
+        assert self.nc.filepath() == str(self.netcdf_file)
```

Comment:
I was going to send a test like that but, because these tests are running on `Travis-ci` with a modern GLIBC, the problem won't surface.  (But more testing never hurt anyone.)

I am building against the [issue506 branch]((https://github.com/ocefpaf/staged-recipes/blob/netcdf4/recipes/netcdf4/meta.yaml#L6-L7)) and the problem [still exists](https://circleci.com/gh/conda-forge/staged-recipes/151) :unamused: